jmx/plugins/org.jboss.tools.jmx.core/src/org/jboss/tools/jmx/core/tree/ObjectNameNode.java

```diff
@@ -62,17 +77,24 @@ public void run(MBeanServerConnection mbsc) throws Exception {
 		    	});
         	}
     	} catch( Exception e ) {
-    		// TODO FIX THIS
+    		JMXActivator.getDefault().getLog().log(new Status(IStatus.ERROR, JMXActivator.PLUGIN_ID, "Error loading object name details", e));
+    		loadError = e;
     	}
     	wrapper = array[0];
     }
 
-
+    public boolean hasLoadError() {
+    	return loadError != null;
+    }
+    
     public ObjectName getObjectName() {
         return on;
     }
 
-    public MBeanInfoWrapper getMbeanInfoWrapper() {
+    public synchronized MBeanInfoWrapper getMbeanInfoWrapper() {
+    	if( wrapper == null ) {
```

Comment:
refresh currently refreshes the entire connection. There's an API in there to make it a bit more incremental but it's currently unused.  We can look into that in the future. 

I've updated the viewers to have a isLoaded() method and efficiency is now much improved. Viewer will  now also show Loading... nodes.